extension/onboarding/onboardingController.jsx

```diff
@@ -0,0 +1,67 @@
+/* globals React, ReactDOM, settings */
+
+this.onboardingController = (function() {
+  const exports = {};
+  const { useState, useEffect } = React;
+  const onboardingContainer = document.getElementById("onboarding-container");
+  let isInitialized = false;
+  let userSettings;
+
+  exports.OnboardingController = function() {
+    const [optinViewAlreadyShown, setOptinViewShown] = useState(true);
+    const [permissionError, setPermissionError] = useState(null);
+
+    useEffect(() => {
+      if (!isInitialized) {
+        isInitialized = true;
+        init();
+      }
+    });
+
+    const init = async () => {
+      userSettings = await settings.getSettings();
+      setOptinViewShown(userSettings.collectTranscriptsOptinShown);
+
+      if (optinViewAlreadyShown) {
+        launchPermission();
+      }
+    };
+
+    const setOptinValue = async value => {
+      userSettings.collectTranscriptsOptinShown = true;
```

Comment:
I think "answered" might be a better term than "shown" (since it can show more than once). Also this should be set to `Date.now()` (to record the date) instead of just a boolean.

Reply:
I've fixed this in  #661